core/src/conn.rs

```diff
@@ -35,17 +38,20 @@ impl MongoConnection {
     /// The initial operation time if provided should come from and will take precedence over the
     /// setting specified in the uri if any.
     pub fn connect(
-        mut client_options: ClientOptions,
+        mut options: (ClientOptions, Option<ConnectionString>),
         current_db: Option<String>,
         operation_timeout: Option<u32>,
         login_timeout: Option<u32>,
     ) -> Result<Self> {
-        client_options.connect_timeout = login_timeout.map(|to| Duration::new(to as u64, 0));
-        let client = Client::with_options(client_options).map_err(Error::InvalidClientOptions)?;
+        options.0.connect_timeout = login_timeout.map(|to| Duration::new(to as u64, 0));
+        let client =
+            Client::with_options(options.0.clone()).map_err(Error::InvalidClientOptions)?;
+        let uuid_repr = options.1.and_then(|opts| opts.uuid_representation);
```

Comment:
Looks like in the `to_json` implementation `for Bson` below, we choose a default `UuidRepresentation` if this is `None`. I wonder if we should instead set that default value right here, storing a `UuidRepresentation` directly instead of storing an `Option<UuidRepresentation>`. I understand this would break from the convention we have for the rest of the fields in `MongoConnection`, so if you don't think we should change it that's ok with me.

Reply:
I had initially done this! I ended up reverting it because all of our data tests instantiate a "connection" without a mongo connection. Thus, there's nothing to read there. So rather than having a default defined here, then defining a "default" if the connection doesn't exist, I opted to let the default be defined right at the end and just pass options through.

Thoughts?